src/components/RegisterModal/RegisterModal.jsx

```diff
@@ -0,0 +1,107 @@
+import ModalWithForm from "../ModalWithForm/ModalWithForm";
+import './RegisterModal.css';
+import useForm from "../../hooks/useForm";
+
+export default function RegisterModal({
+   isOpen,
+  activeModal,
+  onClose,
+  handleSignUp,
+  onSuccessfulSignUpModal,
+  onSignInClick
+}) {
+   const defaultValues = {
+    email: "",
+    password: "",
+    username: "",
+  };
+
+  const { values, isValid, errors, resetForm, handleChange } =
+    useForm(defaultValues);
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (isValid) {
+      handleSignUp(values);
+    }
+    resetForm(defaultValues);
```

Comment:
You need to **reset** the form only when you **open** it or better in a `then` block in order not to do that immediately after `submission` because when there is an error, the modal will still be opened, and the user can try to send the data again to the server without having to type the data into the inputs once more.

Here is an example of how to clear the inputs when you **open** the modal:

```js
  useEffect(() => {
    if (isOpen) {
      resetForm(defaultValues);
    }
  }, [isOpen]); // watch the opening state

```